cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +105,300 @@ def unset_default_org(self):
+        dir_path = (
+            Path(self.project_local_dir) if project else Path(self.global_config_dir)
+        )
+        with open(dir_path / DEFAULT_SERVICES_FILENAME, "w") as f:
+            f.write(json.dumps(self._default_services))
+
+    def _create_default_service_files(self) -> None:
+        """
+        Generate the DEFAULT_SERVICES.json files at global and project scopes.
+
+        @param local_proj_path: should be the local_project_path for the project
+        """
+        global_default_service_file = Path(
+            f"{self.global_config_dir}/{DEFAULT_SERVICES_FILENAME}"
+        )
+        if global_default_service_file.is_file():
+            return
+
+        self._write_default_services_for_dir(self.global_config_dir)
+        for local_proj_dir in self._iter_local_project_dirs():
+            self._write_default_services_for_dir(local_proj_dir)
+
+    def _write_default_services_for_dir(self, dir_path: str) -> None:
+        """Look through the given dir and set all .service files
+        present as the defaults for their given types, and write these
+        out to a DEFAULT_SERVICES.json file in the directory. This occurs
+        once before .service files are migrated to the appropriate
+        services/ sub-directory, so we set the default to the alias
+        that will be assigned during migration.
+
+        @param dir_path: the directory to look through and write
+        the DEFAULT_SERVICES.json file in.
+        """
+        if dir_path is None:
+            return  # pragma: no cover
+
+        dir_path = Path(dir_path)
+        default_services = {}
+        for item in dir_path.iterdir():
+            if item.suffix == ".service":
+                service_type = item.name.replace(".service", "")
+                scope = "global" if item.parent.name == ".cumulusci" else "project"
+                default_services[service_type] = f"{service_type}__{scope}"
+
+        with open(dir_path / DEFAULT_SERVICES_FILENAME, "w") as f:
+            f.write(json.dumps(default_services))
+
+    def _create_services_dir_structure(self) -> None:
+        """
+        Ensure the 'services' directory structure exists.
+        The services dir has the following structure and lives
+        in the global_config_dir:
+
+        services
+        |-- github
+        |   |-- alias1.service
+        |   |-- alias2.service
+        |   |-- ...
+        |-- devhub
+        |   |-- alias1.service
+        |   |-- alias2.service
+        |   |-- ...
+        .
+
+        This also has the advantage that when a new service type
+        is added to cumulusci.yml a new directory for that service type
+        will be created.
+        """
+        services_dir_path = Path(f"{self.global_config_dir}/services")
+        if not Path.is_dir(services_dir_path):
+            Path.mkdir(services_dir_path)
+
+        configured_service_types = self.project_config.config["services"].keys()
```

Comment:
You don't actually need .keys() if you're just going to iterate over it. Iterating over a dict gives you the dict keys. Some people like to explicitly write .keys() though to make it clearer, and that's fine.

Reply:
I'll leave it for now. I think the variable name better conveys what the keys pertain to in this context.